Birthday-2024-Project/Scripts/PieceLogic.gd

```diff
@@ -252,18 +259,19 @@ func _SetReturnPoint():
 func _input(event):
 	if event.is_action_pressed("GrabPiece"):
 		# Check if the user has clicked on the piece's exact shape
-		if _check_shape_clicked():
+		var clicked_cell : int = _check_shape_clicked()
+		if clicked_cell != -1:
 			print("Piece shape clicked: " + name)
-			on_clicked()
+			on_clicked(clicked_cell)
 
-func _check_shape_clicked() -> bool:
+func _check_shape_clicked() -> int:
 	var relative_click_position : Vector2 = get_global_mouse_position() - GetOriginCellPosition()
 	for i in range(_current_cells.size()):
 		var cell00 : Vector2 = Vector2((_current_cells[i].x - 0.5) * levelGridReference.tile_set.tile_size.x, (_current_cells[i].y - 0.5) * levelGridReference.tile_set.tile_size.y)
 		var cell11 : Vector2 = cell00 + Vector2(levelGridReference.tile_set.tile_size.x, levelGridReference.tile_set.tile_size.y)
 		if relative_click_position.x >= cell00.x and relative_click_position.x <= cell11.x and relative_click_position.y >= cell00.y and relative_click_position.y < cell11.y: # I really wish gdscript let you break statements across lines
-			return true # Mouse is inside one of the piece's cells
-	return false
+			return i # Mouse is inside one of the piece's cells
+	return -1
```

Comment:
same case as with 0.1, have it as constant value NO_CELL or something that would have similar meaning